docs/source/developer-docs.rst

```diff
@@ -0,0 +1,69 @@
+Developer Documentation
+===========================
+
+.. toctree::
+   :maxdepth: 2
+
+System Overview
+^^^^^^^^^^^^^^^
+
+The standard Jupyter notebook has three components, (1) the client that makes up the notebook cells, (2) a web server that lists notebook files, directories and serves notebook assets like HTML and CSS (3) a kernel that executes commands in the chosen language (in our case Python).
+
+.. figure:: https://docs.google.com/drawings/d/1Zss2O5rXANNYnaooZhnKhWCVu6wjURmBWNeKLL_QXXc/pub?w=960&h=720
+   :align: center
+
+   Components of the Jupyter Notebook
+
+When you execute python in a Jupyter Notebook cell the python code was transmitted (as a string) over a web socket to the webserver, then proxied through ZeroMQ to the IPykernel where it was evaluated as a Python expression. The results of that execution are then sent back over the ZeroMQ/Websocket connection and displayed in your browser. This is the standard way in which Jupyter notebook takes code from a web browser, and executes it in an interactive shell (kernel) and returns the output.
+
+
+*TODO introduce comm channels*
```

Comment:
My slight preference to put them inside an issue with a checklist if it is not too much work but I am okay with this as it is (if you want to go ahead and merge it).